holidays/countries/fiji.py

```diff
@@ -0,0 +1,174 @@
+from holidays.calendars.gregorian import SEP, OCT, NOV, DEC
+from holidays.constants import PUBLIC, WORKDAY
+from holidays.groups import (
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+)
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    SAT_SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON_TUE,
+    ALL_TO_NEAREST_MON,
+)
+
+
+class Fiji(
+    ObservedHolidayBase,
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+):
+    """
+    References:
+    - https://laws.gov.fj/Acts/DisplayAct/2910#
+    - https://laws.gov.fj/LawsAsMade
+    - https://www.fiji.gov.fj/About-Fiji/Public-Holidays
+    - https://www.timeanddate.com/holidays/fiji/
+    - https://en.wikipedia.org/wiki/List_of_festivals_in_Fiji
+    - https://www.rnz.co.nz/international/pacific-news/249514/new-public-holiday-for-fiji
+    - https://www.fijitimes.com.fj/constitution-day-public-holiday-removed-cabinet/
+    - https://fijivillage.com/news/National-Sports-Day-celebrated-5krs29/
+    - https://fijivillage.com/news/Cabinet-approves-Ratu-Sir-Lala-Sukuna-Day-and-Girmit-Day-and-removes-Constitution-Day-as-a-public-holiday-f48r5x/
+
+    Official Fiji Public Holidays Calendar:
+    - `2016 <https://www.fiji.gov.fj/Media-Center/Press-Releases/GOVERNMENT-APPROVES-2016-PUBLIC-HOLIDAYS.aspx>`_
+    - `2017 <https://www.fiji.gov.fj/Media-Centre/News/GOVERNMENT-APPROVES-2017-PUBLIC-HOLIDAYS>`_
+    - `2018 <https://web.archive.org/web/20180727205733/http://www.employment.gov.fj/images/Laws/Press%20Release%20-%20Government%20Approves%202018%20Public%20Holidays.pdf>`_
+    - `2019 <https://web.archive.org/web/20191018023027/https://www.fiji.gov.fj/About-Fiji/Public-Holidays>`_
+    - `2020 <https://web.archive.org/web/20210103183942/https://www.fiji.gov.fj/About-Fiji/Public-Holidays>`_
+    - `2021-2022 <https://web.archive.org/web/20221223004409/https://www.fiji.gov.fj/About-Fiji/Public-Holidays>`_
+    - `2023 <https://web.archive.org/web/20231129154609/https://www.fiji.gov.fj/About-Fiji/Public-Holidays>`_
+    -` 2024 <https://web.archive.org/web/20250121185434/https://www.fiji.gov.fj/About-Fiji/Public-Holidays>`_
+    - `2025 <https://web.archive.org/web/20250318092311/https://www.fiji.gov.fj/About-Fiji/Public-Holidays>`_
+
+    """
+
+    country = "FJ"
+    supported_categories = (PUBLIC, WORKDAY)
+    # %s (estimated).
+    estimated_label = "%s (estimated)"
+    # %s (observed).
+    observed_label = "%s (observed)"
+    # %s (observed, estimated).
+    observed_estimated_label = "%s (observed, estimated)"
+    # Act No. 13 of 2015
+    start_year = 2016
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self, cls=FijiHinduHolidays, show_estimated=True)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self, cls=FijiIslamicHolidays, show_estimated=True)
```

Comment:
Regarding @KJhellico 's earlier suggestion, this one should work now 👍 
```suggestion
    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
        """
        :param islamic_show_estimated:
            Whether to add "estimated" label to Islamic holidays name if holiday date is estimated.
        """
        ChristianHolidays.__init__(self)
        HinduCalendarHolidays.__init__(self, cls=FijiHinduHolidays, show_estimated=True)
        InternationalHolidays.__init__(self)
        IslamicHolidays.__init__(
            self, cls=FijiIslamicHolidays, show_estimated=islamic_show_estimated
        )
```

Reply:
Isn't this the same code as @KJhellico's implementation?

Reply:
Nvm, just notice that it's indeed exactly that - though I've confirmed that it's working properly on my PC

Reply:
It still creates a blank snapshot with {}, which causes the tests to fail. As a result, the problem persists.

Reply:
Try deleting your local copy then recloning the repo to your PC again - the issue might stem from the files that exist on your local environment but has been `.gitignore`-ed.

I did this all the time myself and it solves ~90% of the weird issues like this.

Reply:
Thanks for the suggestion! I’ve tried deleting my local copy and recloning the repo twice myself for this case, but it didn’t help me at all. Any other ideas for troubleshooting this?

Reply:
1. Try this:
```python
from holidays.version import __version__
print(__version__)  # should get v0.70 here
```
If it's v0.69 or below then you might have accidentally run codes against earlier version of `holidays` installed somewhere else on your PC.

2. Were there any other errors while running `make test`? If they do, post them here 🙏

3. If all else fails, list out OS and Python version here for now

Reply:
Sorry for my silly mistakes—I wasn’t even able to incorporate a minor suggestion into my PR and ended up increasing both of your workloads @KJhellico @PPsyrius. As you asked for the failing tests, here they are:-
```bash
FAILED tests/countries/test_fiji.py::TestFiji::test_2019 - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_2020 - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_2021 - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_2022 - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_2023 - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_2024 - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_2025 - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_boxing_day - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_christmas_day - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_constitution_day - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_diwali - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_easter_monday - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_easter_saturday - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_fiji_day - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_girmit_day - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_good_friday - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_national_sports_day - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_new_years_day - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_no_holidays - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_prophets_birthday - AttributeError: 'Fiji' object has no attribute 'observed'
FAILED tests/countries/test_fiji.py::TestFiji::test_ratu_sir_lala_sukuna_day - AttributeError: 'Fiji' object has no attribute 'observed'

Reply:
I've just cloned your repo, checkout to `add-fiji-holidays`, made suggested changes, and the test is running OK, and snapshot also not empty. 🤷‍♂️

Reply:
And on the `dev` branch, is it running fine for you?

Reply:
Yes, I cloned the `dev` branch and ran `make check`—everything went fine there. Also confirmed that the `add-fiji-holidays` branch is using `v0.70`.